src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodOrUserDefinedOperatorSymbol.cs

```diff
@@ -46,6 +46,7 @@ public sealed override bool ReturnsVoid
             this.SetReturnsVoid(_lazyReturnType.IsVoidType());
 
             this.CheckEffectiveAccessibility(_lazyReturnType, _lazyParameters, diagnostics);
+            this.CheckFileTypeUsage(_lazyReturnType, _lazyParameters, diagnostics);
```

Comment:
Can this new logic be folded into `CheckEffectiveAccessibility`?

Reply:
I don't really think so. We need to run the file types checks even on private methods, so there's not a simple place to slot these checks into the existing `CheckEffectiveAccessibility` code. We could introduce a helper which calls both methods sequentially, but this has only two usages, so I'm leaning against that.